File: eMovieFinder/eMovieFinder.Model/Dtos/Requests/User/UserUpdateRequest.cs
using System.Collections.Generic;
using System.ComponentModel;

namespace eMovieFinder.Model.Dtos.Requests.User
{
    public class UserUpdateRequest
    {
        public string? Username { get; set; }
        public string? Email { get; set; }
        public string? Password { get; set; }
        public List<string>? Roles { get; set; }
        public string? ImagePlainText { get; set; }
        [DefaultValue(null)]
        public byte[]? Image { get; set; }
        public bool IsLockoutEnabled { get; set; }
    }
}

File: eMovieFinder-UI/emovie_finder_desktop/lib/src/models/dtos/requests/user/user_update_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'user_update_request.g.dart';

@JsonSerializable()
class UserUpdateRequest {
  final String? username;
  final String? email;
  final String? password;
  final dynamic roles;
  final String? imagePlainText;
  final bool? isLockoutEnabled;

  UserUpdateRequest({
    this.username,
    this.email,
    this.password,
    this.roles,
    this.imagePlainText,
    this.isLockoutEnabled
  });

  factory UserUpdateRequest.fromJson(Map<String, dynamic> json) => _$UserUpdateRequestFromJson(json);
  Map<String, dynamic> toJson() => _$UserUpdateRequestToJson(this);
}